### PR TITLE
test: Build source package in VM

### PR DESCRIPTION
**Pull request by mvollmer**
This avoids using OS specific tools on the testing host.

Note that we don't spawn a dedicated VM just to build the source package, this happens as part of vm-build and vm-create runs.

### CODE REVIEW

**Comment by stefwalter** [comment]
I don't think this will work  for @sgallagher who uses make-srpms as a quick way to get the tree to build elsewhere.

**Comment by mvollmer** [comment]
We can keep tools/make-srpm, of course.

File: test/vm-build [hunk 1]
```diff
@@ -31,20 +31,17 @@ args = parser.parse_args()
 
 def build(machine):
     """Build Cockpit in a virtual machine"""
-    srpm = subprocess.check_output([ "../tools/make-srpm" ]).strip()
+    source = subprocess.check_output([ "../tools/make-tar" ]).strip()
```
Comment (stefwalter) [inline test/vm-build]: Can we call this script 'make-source' to match the nomenclature you've used throughout all the other various and stuff. It also helps distinguish it from a real tarball (which is created via 'make dist').

File: test/vm-build [hunk 2]
```diff
@@ -31,20 +31,17 @@ args = parser.parse_args()
...
-        if args.verbose:
-            mock_opts += " --verbose"
-        if args.quick:
-            mock_opts += " --nocheck --define='selinux 0'"
-        mock_cmd = ("mock --offline --no-clean --resultdir /var/tmp/mock %s --rebuild /var/tmp/%s"
-                    % (mock_opts, os.path.basename(srpm)))
-        machine.execute('su - builder -c "%s"' % mock_cmd)
-        machine.download_dir("/var/tmp/mock", "mock")
+        machine.upload([ "guest/lib" ], "/usr/lib/testvm")
+        machine.upload([ source, "./guest/%s.build-source" % machine.os ], "/var/tmp")
```
Comment (stefwalter) [inline test/vm-build]: Why does this need an OS specific name?

File: tools/make-srpm [hunk 3]
```diff
@@ -26,45 +26,9 @@ fi
 base=$(cd $(dirname $0); pwd -P)
 rpm_tmpdir=$(mktemp -d -t rpmbuild.XXXXXX)
 
-# First make a .tar.gz with the source code
-(
-    cd $base/..
+srpm=$(cd ${rpm_tmpdir} && "$base/../test/guest/lib/make-srpm" $("$base/make-tar"))
```
Comment (stefwalter) [inline tools/make-srpm]: I would suggest disentangling this a bit, and calling make-tar (or make-source) on a prior line.

**Comment by mvollmer** [comment]
The build-results are not copied when the build fails.

**Comment by mvollmer** [comment]
I think I'll merge vm-build into vm-install here.

**Comment by stefwalter** [comment]
> I think I'll merge vm-build into vm-install here.

Makes sense

**Comment by mvollmer** [comment]
> The build-results are not copied when the build fails.

Fixed.

File: test/vm-build (hunk 1 above)
Comment (mvollmer) [inline test/vm-build]: Done.

File: test/vm-build (hunk 2 above)
Comment (mvollmer) [inline test/vm-build]: The script will be different for different OSes, like Debian.

File: tools/make-srpm (hunk 3 above)
Comment (mvollmer) [inline tools/make-srpm]: Done.

**Comment by mvollmer** [comment]
> I think I'll merge vm-build into vm-install here.

Done, but fedora-atomic is not yet supported.

**Comment by mvollmer** [comment]
New fedora-testing image is missing.

**Comment by mvollmer** [comment]
> New fedora-testing image is missing.

Uploaded.

**Comment by mvollmer** [comment]
> Done, but fedora-atomic is not yet supported.

Fixed.

**Comment by mvollmer** [comment]
The fedora-atomic and fedora-testing failures are very likely unrelated.  Building and installing seem to work for both configurations.

**Comment by stefwalter** [comment]
Needs rebase.

**Comment by mvollmer** [comment]
> Needs rebase.

Done.

**Comment by akilawickey** [comment]
i got the following when i run cockpit on fedora 23 vm

[akilawickey@asak-opensource cockpit]$ sudo vagrant destroy
==> default: Remove stale volume...
==> default: Domain is not created. Please run `vagrant up` first.
[akilawickey@asak-opensource cockpit]$ sudo vagrant up
Bringing machine 'default' up with 'libvirt' provider...
==> default: Creating image (snapshot of base box volume).
==> default: Creating domain with the following settings...
==> default:  -- Name:              cockpit_default
==> default:  -- Domain type:       kvm
==> default:  -- Cpus:              1
==> default:  -- Memory:            1024M
==> default:  -- Management MAC:    
==> default:  -- Loader:            
==> default:  -- Base box:          fedora-23-cloud-base
==> default:  -- Storage pool:      default
==> default:  -- Image:             /var/lib/libvirt/images/cockpit_default.img (41G)
==> default:  -- Volume Cache:      default
==> default:  -- Kernel:            
==> default:  -- Initrd:            
==> default:  -- Graphics Type:     vnc
==> default:  -- Graphics Port:     5900
==> default:  -- Graphics IP:       127.0.0.1
==> default:  -- Graphics Password: Not defined
==> default:  -- Video Type:        cirrus
==> default:  -- Video VRAM:        9216
==> default:  -- Keymap:            en-us
==> default:  -- INPUT:             type=mouse, bus=ps2
==> default:  -- Command line : 
Error while creating domain: Error saving the server: Call to virDomainDefineXML failed: invalid argument: could not find capabilities for domaintype=kvm

**Comment by stefwalter** [comment]
@akilawickey This is an odd place to startup a discussion. In an old unrelated bug. 

That said, you probably need to install qemu-kvm and qemu-system-x86_64

**Pull request closed without merging**